src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Protocol/RazorFoldingRangeRequestParam.cs

```diff
@@ -10,4 +10,7 @@ internal class RazorFoldingRangeRequestParam : FoldingRangeParams
 {
     [DataMember(Name = "hostDocumentVersion")]
     public int HostDocumentVersion { get; init; }
+
+    [DataMember(Name = "projectContext")]
+    public VSProjectContext? ProjectContext { get; init; }
```

Comment:
Okay, I looked around the protocol assemblies, and I think maybe this isn't actually necessary. Sorry, I think I send you down the garden path. `FoldingRangeParams` has a `TextDocumentIdentifier` but the platform provides a JSON converter, that will put a `VSTextDocumentIdentifier` in its place, which has a project context on it. I _think_ we can just ignore this, and it should flow all the way through to Roslyn all by itself.

Seems to be the case for me anyway:
<img width="1001" alt="image" src="https://github.com/dotnet/razor/assets/754264/a1d55d76-82cf-40ab-b99b-7a0d1f15057a">

Reply:
Yea this should be correct. Will revert